weldx/core.py

```diff
@@ -270,8 +270,9 @@ class TimeSeries:
     def __init__(
         self,
         data: Union[pint.Quantity, MathematicalExpression],
-        time: Union[None, pd.TimedeltaIndex, pint.Quantity] = None,
+        time: Union[types_time_like, Time] = None,
```

Comment:
This is totally unrelated to this PR, but we should find a solution without ring dependencies to include `Time` into `types_time_like` and other fitting time type definitions

Reply:
cant we just define `types_time_like` (and other time related types) in the `.time` module would also make sense to put it there imo

Reply:
Yes, that's a good option I also thought of. But I wasn't sure about it since we have the dedicated types module. Still not sure how to add `Time` to `types_time_like` while using it all over the place in the classes type hints. Does Python have some sort of forward declaration (will check that)?

Reply:
Done in PR #468